### PR TITLE
use TSLint for JS files too

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- use TSLint 4.x to lint JS files too! (mostly object alph)
- remove eslint

### CODE REVIEW

File: gulp/util/text.js [hunk 1]
```diff
@@ -49,19 +49,19 @@ renderer.code = (textContent, language) => {
...
     fileStream: (filename, contents) => {
         const stream = strSource(filename);
         stream.end(contents);
         return stream;
     },
+
+    // synchronously read and return string content of file.
+    fromFile: (filepath) => fs.readFileSync(filepath, "utf8"),
+
+    highlight: highlight,
```
Comment (adidahiya) [inline gulp/util/text.js]: this should be flagged as a failure by `object-literal-shorthand`... what's going on?

File: gulp/karma.js [hunk 2]
```diff
@@ -91,16 +91,16 @@ module.exports = (gulp, plugins, blueprint) => {
 
         gulp.task(`karma-unit-${project.id}`, (done) => {
             const config = Object.assign(createConfig(project), {
-                reporters: [ "mocha" ],
-                singleRun: false,
+                browsers: ["Chrome"],
```
Comment (adidahiya) [inline gulp/karma.js]: nit: inconsistent spacing inside `[ ]`

File: gulp/util/text.js (hunk 1 above)
Comment (giladgray) [inline gulp/util/text.js]: huh that rule had to be explicitly enabled here

Comment (adidahiya) [inline gulp/util/text.js]: weird, it's enabled in the recommended config: https://github.com/palantir/tslint/blob/47212df6d3e97eb1e2da6ed0d1fb43cda4e32126/src/configs/recommended.ts#L79

File: gulp/karma.js (hunk 2 above)
Comment (giladgray) [inline gulp/karma.js]: fixed

**Pull request merged**